File: packages/http-client-csharp-generator/src/TypeSpec.Extension/Emitter.Csharp/src/type/operationResponse.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import { BodyMediaType } from "./bodyMediaType.js";
import { HttpResponseHeader } from "./httpResponseHeader.js";
import { InputType } from "./inputType.js";

export interface OperationResponse {
    StatusCodes: number[];
    BodyType?: InputType;
    BodyMediaType: BodyMediaType;
    Headers: HttpResponseHeader[];
    ContentTypes?: string[];
    IsErrorResponse: boolean;
}

File: packages/http-client-csharp-generator/src/AutoRest.CSharp/Common/Input/InputTypes/OperationResponse.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;

namespace AutoRest.CSharp.Common.Input;

internal record OperationResponse(IReadOnlyList<int> StatusCodes, InputType? BodyType, BodyMediaType BodyMediaType, IReadOnlyList<OperationResponseHeader> Headers, bool IsErrorResponse, IReadOnlyList<string> ContentTypes)
{
    public OperationResponse() : this(StatusCodes: Array.Empty<int>(), BodyType: null, BodyMediaType: BodyMediaType.None, Headers: Array.Empty<OperationResponseHeader>(), IsErrorResponse: false, ContentTypes: Array.Empty<string>()) { }
}
